packages/@webex/internal-plugin-metrics/src/business-metrics.ts

```diff
@@ -62,6 +62,32 @@ export default class BusinessMetrics extends GenericMetrics {
     return this.submitEvent({kind: 'buisness-events:business_metrics -> ', name, event});
   }
 
+  /**
+   * Submit a buisness metric to our metrics endpoint, going to the business_metrics_wxcc_desktop table
+   * all event payload keys are converted into a hex string value
+   * @param {string} name of the metric
+   * @param {EventPayload} payload payload of the metric
+   * @returns {Promise<any>}
+   */
+  private submitWxccDesktopEvent({name, payload}: {name: string; payload: EventPayload}) {
```

Comment:
It seems to me that we shouldnt have product specific APIs. Can we achieve the same in a generic way? eg some "transform" mechanism that we can inject, or by setting config centrally.

Reply:
The API the SDK currently exposes is 
```
submitBusinessEvent({
    name: string;
    payload: EventPayload;
    table?: Table;
})
```
(we can't make a breaking change or else we have to bump the major version number of the SDK)

We could add a second method like:
```
submitBusinessEvent({
    name: string;
    payload: EventPayload;
    eventMetadata: EventPayload;
})
```
where `eventMetadata` contains the `appType` and `indexHint`, or anything else, for future generic use